integrationTests/singleShard/block/executingMiniblocksSc_test.go

```diff
@@ -148,57 +232,396 @@ func TestProcessesJoinGameOf100PlayersRewardAndEndgame(t *testing.T) {
+	numOfNodes := 3
+	maxShards := uint32(2)
+	nodes := make([]*integrationTests.TestProcessorNode, numOfNodes)
+	nodes[0] = integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
+	nodes[1] = integrationTests.NewTestProcessorNode(maxShards, 1, 1, advertiserAddr)
+	nodes[2] = integrationTests.NewTestProcessorNode(maxShards, sharding.MetachainShardId, 1, advertiserAddr)
+
+	idxProposer := 0
+	nrPlayers := 100
+	players := make([]*integrationTests.TestWalletAccount, nrPlayers)
+	for i := 1; i < nrPlayers; i++ {
+		players[i] = integrationTests.CreateTestWalletAccount(nodes[idxProposer].ShardCoordinator, 0)
+	}
+
+	idxProposers := make([]int, 3)
+	idxProposers[0] = idxProposer
+	idxProposers[1] = 1
+	idxProposers[2] = 2
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
 		}
-		time.Sleep(time.Second)
+	}()
 
-		startTime := time.Now()
-		proposeBlock(nodes, idxProposer, round)
-		elapsedTime := time.Since(startTime)
-		fmt.Printf("Block Created in %s\n", elapsedTime)
-		round = incrementAndPrintRound(round)
+	for _, n := range nodes {
+		_ = n.Messenger.Bootstrap()
+	}
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(time.Second)
+
+	round := uint64(0)
+	round = incrementAndPrintRound(round)
+
+	hardCodedSk, _ := hex.DecodeString("5561d28b0d89fa425bbbf9e49a018b5d1e4a462c03d2efce60faf9ddece2af06")
+	hardCodedScResultingAddress, _ := hex.DecodeString("000000000000000000005fed9c659422cd8429ce92f8973bba2a9fb51e0eb3a1")
+	nodes[idxProposer].LoadTxSignSkBytes(hardCodedSk)
 
-		nodeCallsRewardAndSend(nodes, idxProposer, idxProposer, withdrawValue, rr, hardCodedScResultingAddress)
-		nodeEndGame(nodes, idxProposer, rr, hardCodedScResultingAddress)
+	initialVal := big.NewInt(10000000)
+	topUpValue := big.NewInt(500)
+	integrationTests.MintAllNodes(nodes, initialVal)
+	integrationTests.MintAllPlayers(nodes, players, initialVal)
+
+	nrRoundsToPropagateMultiShard := 5
+	if maxShards == 1 {
+		nrRoundsToPropagateMultiShard = 1
+	}
+
+	for i := 0; i < nrRoundsToPropagateMultiShard; i++ {
+		time.Sleep(time.Second)
+		deployScTx(nodes, idxProposer, string(scCode))
 		time.Sleep(time.Second)
+		proposeBlock(nodes, idxProposers, round)
+		time.Sleep(time.Second)
+		syncBlock(t, nodes, idxProposers, round)
+		round = incrementAndPrintRound(round)
+	}
+
+	runMultipleRoundsOfTheGame(
+		t,
+		100,
+		nrPlayers,
+		nodes,
+		players,
+		topUpValue,
+		hardCodedScResultingAddress,
+		round,
+		maxShards,
+		idxProposers,
+	)
+
+	checkRootHashes(t, nodes, []int{0})
 
-		startTime = time.Now()
-		proposeBlock(nodes, idxProposer, round)
-		elapsedTime = time.Since(startTime)
-		fmt.Printf("Block Created in %s\n", elapsedTime)
+	time.Sleep(time.Second)
+}
+
+func TestProcessesJoinGame100PlayersMultipleTimesRewardAndEndgameInMultipleRoundsMultiShardMultiNode(t *testing.T) {
+	t.Skip("this is a stress test for VM and AGAR.IO")
+
+	stepDelay = time.Nanosecond
```

Comment:
Is this enough? Also, should not rewrite the stepDelay var.

Reply:
stepDelay is used for every transaction dispatching, I put that to 1 nanosecond, than have a bigger stepDelay between transaction dispacthing to block proposing and to sync. Otherwise it takes 100seconds to dispatch 100 transactions.